### PR TITLE
TimeseriesSearchAndSelect component (TagSearchAndSelect)

### PR DESCRIPTION
**Pull request by follan**

### CODE REVIEW

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-81.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-81.gearboxjs.preview.cogniteapp.com

File: src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+import Adapter from 'enzyme-adapter-react-16';
+import Mock = jest.Mock;
+
+configure({ adapter: new Adapter() });
+
+sdk.TimeSeries.search = jest.fn();
+sdk.Assets.list = jest.fn();
+
+const { Search } = Input;
+const propsCallbacks: { [name: string]: Mock } = {
```
Comment (f1cognite) [inline src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx]: Do you need to specify the type here? I would just remove `: { [name: string]: Mock }`

File: src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+afterEach(() => {
+  Object.keys(propsCallbacks).forEach((key: string) =>
+    propsCallbacks[key].mockClear()
+  );
+  // @ts-ignore
+  sdk.TimeSeries.search.mockClear();
+});
+
+
+describe('TimeserieSearchAndSelect', () => { // tslint:disable-line
```
Comment (f1cognite) [inline src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx]: Why `tslint:disable-line`?

File: src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+  rootAsset?: VId;
+  filterRule?: (timeseries: sdk.Timeseries) => boolean;
+  onError?: (error: Error) => void;
+}
+
+export interface TimeserieSearchAndSelectState {
+  assetId?: VId;
+  assets: VAsset[];
+  fetching: boolean;
+  searchResults: any;
```
Comment (f1cognite) [inline src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx]: Should we have a propper type here?

File: src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+  lastFetchId: number;
+}
+
+class TimeserieSearchAndSelect extends React.Component<
+  TimeserieSearchAndSelectProps,
+  TimeserieSearchAndSelectState
+> {
+  static defaultProps = {
+    selectedTimeseries: [],
+    filterRule: (_: sdk.Timeseries) => true,
```
Comment (f1cognite) [inline src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx]:
```suggestion
    filterRule: () => true,
```

File: src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    });
+  }
+
+  async componentDidMount() {
+    const assets = await sdk.Assets.list({ depth: 0 });
+    this.setState({ assets: assets.items });
+  }
+
+  onSelectAsset = (assetId: VId): void => {
+    console.log('102');
```
Comment (f1cognite) [inline src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx]: Remove this?

File: src/mocks/timeseriesList.ts [hunk 6]
```diff
@@ -0,0 +1,343 @@
+/* tslint:disable */
```
Comment (f1cognite) [inline src/mocks/timeseriesList.ts]: We should avoid disabling an entire file

File: src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx (hunk 2 above)
Comment (follan) [reply src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx]: tslint complains that this function is too many lines, and it should be broken up into more functions. Generally I think that is a good idea, but limiting a describe block in tests seems wrong. Would be better to disable that exact rule, but I haven't been able to find out what the rule is called

Comment (follan) [reply src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx]: Found the rule name. Will update PR

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-81.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-81.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-81.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-81.gearboxjs.preview.cogniteapp.com

File: src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx (hunk 1 above)
Comment (follan) [reply src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx]: Done

File: src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx (hunk 2 above)
Comment (follan) [reply src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.test.tsx]: Done

File: src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx (hunk 3 above)
Comment (follan) [reply src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx]: Fixed

File: src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx (hunk 4 above)
Comment (follan) [reply src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx]: Done

File: src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx (hunk 5 above)
Comment (follan) [reply src/components/TimeseriesSearchAndSelect/TimeseriesSearchAndSelect.tsx]: Done

File: src/mocks/timeseriesList.ts (hunk 6 above)
Comment (follan) [reply src/mocks/timeseriesList.ts]: Changed to disable specific rule

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-81.gearboxjs.preview.cogniteapp.com

File: src/components/DetailCheckbox/DetailCheckbox.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Checkbox } from 'antd';
+
+const Container = styled.div`
+  background: white;
+  border-radius: 3px;
+  position: relative;
+  border: 1px solid rgba(0, 0, 0, 0.2);
```
[the remark this answers is not in the document]
Comment (follan) [reply src/components/DetailCheckbox/DetailCheckbox.tsx]: Updated PR

File: src/components/DetailCheckbox/DetailCheckbox.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+  onContainerClick,
+  title,
+  className,
+  checkable,
+}) => (
+  <Container
+    className={className}
+    onClick={disabled ? null : onContainerClick}
+    style={{
+      borderColor: `${checked ? '#1890ff' : 'rgba(0, 0, 0, 0.2)'}`,
```
[the remark this answers is not in the document]
Comment (follan) [reply src/components/DetailCheckbox/DetailCheckbox.tsx]: Updated PR

File: src/components/DetailCheckbox/DetailCheckbox.tsx [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+      width: '100%',
+      border: 'none',
+      padding: '8px',
+      marginBottom: '4px',
+    }}
+  >
+    <div style={{ wordBreak: 'break-all' }}>
+      <Title
```
[the remark this answers is not in the document]
Comment (follan) [reply src/components/DetailCheckbox/DetailCheckbox.tsx]: Updated PR

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-81.gearboxjs.preview.cogniteapp.com

**Pull request merged**